scripts/alert_emails_report/generate-report.ts

```diff
@@ -0,0 +1,96 @@
+import path from 'path';
+import _ from 'lodash';
+import moment from 'moment';
+import CampaignMonitor from '../alert_emails/campaign-monitor';
+import { getFirestore } from '../alert_emails/firestore';
+import { ALERT_EMAIL_GROUP, toISO8601 } from '../alert_emails/utils';
+import fipsByStateCode from '../what-the-fips/2018-census-fips-codes.json';
+import { findStateByFips, getAllStateFips } from '../../src/common/locations';
+import GoogleSheets from '../common/google-sheets';
+
+const REPORT_SPREADSHEET_ID = '1cs3Wyh8Gda0H18_-x5RYp7AJ3FwB-I1ewAQTBWEk1YY';
+
+interface StateFipsMap {
+  [stateCode: string]: {
+    [countyFips: string]: string;
+  };
+}
+
+const keyFile = path.join(
+  __dirname,
+  '../alert_emails/google-service-account.json',
+);
+
+async function generateAlertEmailReport() {
+  const db = getFirestore();
+  const gsheets = new GoogleSheets(keyFile);
+  await gsheets.authenticate();
+
+  // TODO: Confirm, what is the best way to measure engagement stats?
+  const dateFrom = new Date('2020-07-24'); // initial send
+  const dateTo = moment(dateFrom).add(8, 'day').toDate();
+  const engagementStats = await fetchEngagementStats(dateFrom, dateTo);
+  await gsheets.appendRows(REPORT_SPREADSHEET_ID, 'data_engagement!A2:IE', [
+    [engagementStats],
+  ]);
+
+  const fipsStates = getAllStateFips();
+  const subscriptionStats = await Promise.all(
+    fipsStates.map(stateFips => fetchStateSubscriptionStats(db, stateFips)),
```

Comment:
We would probably like to know about top counties as well...

Like I could imagine that the report has a tab with all states and their subscriber counts, and another tab with all counties (or top 100 if that's too many) with subscriber counts...

Reply:
Updated!